### ISSUE TITLE
Migrate readme project to another docs framework

### ISSUE THREAD

**Issue by armanbilge** [scala-js/scala-js-dom#468]
The readme project currently depends on [scalatex](https://github.com/lihaoyi/scalatex) which unfortunately no longer seems to be maintained. Thus the docs should migrate to another framework, e.g. [mdoc supports Scala.js](https://scalameta.org/mdoc/docs/js.html) although it's not entirely clear to me if this is introducing a circular dependency or not.

**Comment by aappddeevv** [comment]
Just thinking out loud that it may be easier to just make the readme a static github md file and remove interactivity and other visuals. Then just add the readme as the main readme.md for the repo. Maybe add some critical content if someone identifies it is missing. At least reduces complexity for awhile.

**Comment by armanbilge** [comment]
Yeah, that can be our back-up plan if things break before someone gets around to this. For now still seems to be working!

**Comment by armanbilge** [comment]
At least a couple issues have reported issues running `prePR` due to weird scalatex errors. So this is even more incentive to change.

```
[info] compiling 2 Scala sources to C:\Users\mario\code\src\github.com\mabasic\scala-js-dom\readme\target\scala-2.12\classes ...
[error] C:\Users\mario\code\src\github.com\mabasic\scala-js-dom\readme\Index.scalatex:43:6: type mismatch;
[error]  found   : scalatex.site.Section.Proxy
[error]  required: scalatags.Text.all.Frag
[error]     (which expands to)  scalatags.generic.Frag[scalatags.text.Builder,String]
[error] @sect{scala-js-dom}
[error]      ^
[error] one error found
[error] (readme / Compile / compileIncremental) Compilation failed
[error] Total time: 136 s (02:16), completed 31.08.2022. 22:51:29
```

- https://github.com/scala-js/scala-js-dom/pull/724#issuecomment-1233411325
- https://github.com/scala-js/scala-js-dom/pull/628#issuecomment-965546913

**Comment by zetashift** [comment]
What about just using scaladoc 3? https://docs.scala-lang.org/scala3/scaladoc.html

**Comment by armanbilge** [comment]
@zetashift does Scaladoc 3 support Scala.js? I.e. can it be used to make interactive in-browser snippets?

**Comment by zetashift** [comment]
That's a good one, it has runnable snippets, but I can't find anything about it for interative snippets :S
